src/xbot/common/math/PID.java

```diff
@@ -79,6 +116,9 @@ public double calculate(double goal, double current,
             m_result = m_minimumOutput;
         }
         result = m_result;
+        
+        errorIsSmall = Math.abs(m_targetInputValue - m_currentInputValue) < errorTolerance;
```

Comment:
Regardless of whether the current implementation of one caller would _probably_ avoid it, it's faulty logic and fundamentally a bug. I could easily imagine this case coming up in tests or while tuning parameters, and a robot doing unpredictable things is never a good thing. The fix is simple and should be made.